src/main/webapp/app/lecture/pdf-preview/pdf-preview.component.ts

```
@@ -74,386 +73,167 @@
-        if (nextPageIndex > 0 && nextPageIndex <= this.totalPages()) {
-            this.currentPage.set(nextPageIndex);
-            const canvas = this.pdfContainer().nativeElement.querySelectorAll('.pdf-canvas-container canvas')[this.currentPage() - 1] as HTMLCanvasElement;
-            this.updateEnlargedCanvas(canvas);
+            this.attachmentUnitService.update(this.attachmentUnit()!.lecture!.id!, this.attachmentUnit()!.id!, formData, undefined, finalHiddenPages.join(',')).subscribe({
+                next: async () => {
+                    if (finalHiddenPages.length > 0) {
+                        const pdfFileWithHiddenPages = await this.createHiddenVersionOfAttachment(finalHiddenPages);
+
+                        this.attachmentService.getAttachmentByParentAttachmentId(this.attachmentUnit()!.attachment!.id!).subscribe({
+                            next: async (attachmentRes) => {
+                                if (attachmentRes.body && attachmentRes.body.id) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)